src/OpenLoco/src/GameCommands/Track/RemoveTrack.cpp

```diff
@@ -0,0 +1,329 @@
+#include "RemoveTrack.h"
+#include "Economy/Economy.h"
+#include "Map/RoadElement.h"
+#include "Map/SignalElement.h"
+#include "Map/TileManager.h"
+#include "Map/Track/TrackData.h"
+#include "Map/TrackElement.h"
+#include "Objects/BridgeObject.h"
+#include "Objects/TrackExtraObject.h"
+#include "Objects/TrackObject.h"
+#include "Random.h"
+#include "RemoveSignal.h"
+#include "RemoveTrainStation.h"
+#include "Vehicles/Vehicle.h"
+#include "ViewportManager.h"
+
+namespace OpenLoco::GameCommands
+{
+    static World::TrackElement* getElTrackAt(const TrackRemovalArgs& args, const uint8_t trackFlags, const World::Pos3 pos, const uint8_t sequenceIndex)
+    {
+        auto tile = World::TileManager::get(pos);
+        for (auto& el : tile)
+        {
+            auto* elTrack = el.as<World::TrackElement>();
+            if (elTrack == nullptr)
+            {
+                continue;
+            }
+            if (elTrack->rotation() != args.rotation)
+            {
+                continue;
+            }
+            if (elTrack->sequenceIndex() != sequenceIndex)
+            {
+                continue;
+            }
+            if (elTrack->trackObjectId() != args.trackObjectId)
+            {
+                continue;
+            }
+            if ((elTrack->flags() & 0x30) != trackFlags)
```

Comment:
can we just check the flags as normal one after each other rather than use the more cryptic &0x30.
```
if (elTrack->isGhost() != (trackFlags & Flags::isGhost)){continue;}
if (elTrack->isAiAllocated() != (trackFlags & Flags::isAiAllocated)){continue;}
```